inc/umock_c/umock_c_internal.h

```diff
@@ -841,48 +846,80 @@ typedef int(*TRACK_DESTROY_FUNC_TYPE)(PAIRED_HANDLES* paired_handles, const void
         return mock_call_modifier; \
     } \
 
-#define IMPLEMENT_STRICT_EXPECTED_MOCK(return_type, name, ...) \
-    IMPLEMENT_MOCK_FUNCTION(umock_c_strict_expected_, 0, return_type, name, __VA_ARGS__)
+#define IMPLEMENT_STRICT_EXPECTED_MOCK_DECL(return_type, name, ...) \
+    IMPLEMENT_MOCK_FUNCTION_DECL(umock_c_strict_expected_, 0, return_type, name, __VA_ARGS__)
+
+#define IMPLEMENT_STRICT_EXPECTED_MOCK_IMPL(return_type, name, ...) \
+    IMPLEMENT_MOCK_FUNCTION_IMPL(umock_c_strict_expected_, 0, return_type, name, __VA_ARGS__)
+
+#define IMPLEMENT_EXPECTED_MOCK_DECL(return_type, name, ...) \
+    IMPLEMENT_MOCK_FUNCTION_DECL(umock_c_expected_, 1, return_type, name, __VA_ARGS__)
 
-#define IMPLEMENT_EXPECTED_MOCK(return_type, name, ...) \
-    IMPLEMENT_MOCK_FUNCTION(umock_c_expected_, 1, return_type, name, __VA_ARGS__)
+#define IMPLEMENT_EXPECTED_MOCK_IMPL(return_type, name, ...) \
+    IMPLEMENT_MOCK_FUNCTION_IMPL(umock_c_expected_, 1, return_type, name, __VA_ARGS__)
+
```

Comment:
One day we should eliminate the non strict APIs. Some other task, some other fight

Reply:
Created [https://msazure.visualstudio.com/One/_workitems/edit/29596420](https://msazure.visualstudio.com/One/_workitems/edit/29596420)